core/src/test/java/org/elasticsearch/index/translog/TranslogTests.java

```diff
@@ -967,10 +967,32 @@ public void testTranslogWriter() throws IOException {
         final int numOps = randomIntBetween(10, 100);
         byte[] bytes = new byte[4];
         ByteArrayDataOutput out = new ByteArrayDataOutput(bytes);
+        long minSeqNo = Translog.INITIAL_MIN_SEQ_NO;
+        long maxSeqNo = Translog.INITIAL_MAX_SEQ_NO;
+        final Set<Long> seenSeqNos = new HashSet<>();
+        boolean opsHaveValidSequenceNumbers = false;
         for (int i = 0; i < numOps; i++) {
             out.reset(bytes);
             out.writeInt(i);
-            writer.add(new BytesArray(bytes));
+            long seqNo;
+            do {
+                seqNo = opsHaveValidSequenceNumbers ? randomNonNegativeLong() : SequenceNumbersService.UNASSIGNED_SEQ_NO;
+                opsHaveValidSequenceNumbers = opsHaveValidSequenceNumbers || !rarely();
+            } while (seenSeqNos.contains(seqNo));
+            if (seqNo != SequenceNumbersService.UNASSIGNED_SEQ_NO) {
+                seenSeqNos.add(seqNo);
+                if (minSeqNo == Translog.INITIAL_MIN_SEQ_NO) {
```

Comment:
wouldn't it be simpler to calculate those in the end based on seenSeqNos? we can also wrap this whole thing in a opsHaveValidSequenceNumbers chcek and then the inner loop won't need to change for UNASSIGNED_SEQ_NO